src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiImporter.cs

```diff
@@ -0,0 +1,370 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Java.Interop.Tools.JavaTypeSystem.Models;
+
+namespace Java.Interop.Tools.JavaTypeSystem
+{
+	public class JavaXmlApiImporter
+	{
+		public static JavaTypeCollection ParseString (string xml, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Parse (xml, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		public static JavaTypeCollection Parse (TextReader reader, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Load (reader, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		public static JavaTypeCollection Parse (string filename, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Load (filename, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		static JavaTypeCollection Parse (XDocument doc, JavaTypeCollection? collection = null)
+		{
+			collection ??= new JavaTypeCollection ();
+
+			var root = doc.Root;
+
+			if (root is null)
+				throw new Exception ("Invalid XML file");
+
+			collection.ApiSource = root.XGetAttributeOrNull ("api-source");
+			collection.Platform = root.XGetAttributeOrNull ("platform");
+
+			var packages = new List<JavaPackage> ();
+
+			foreach (var elem in root.Elements ()) {
+				switch (elem.Name.LocalName) {
+					case "package":
+						packages.Add (ParsePackage (elem, collection));
```

Comment:
Question: what happens -- vs. what *should* happen -- if the same package is listed multiple times?

```xml
<api>
  <package name="foo" />
  <package name="foo" />
</api>
```

This is unlikely to happen in a single file, but we can imagine a "process multiple separate files at once" construct, in which the same package is listed in more than one file…

Reply:
Ah, looks like this'll hit `JavaTypeCollection.AddPackage()`, which returns a previously encountered package.  Thus, if there *are* multiple packages with the same name, they'll add to the same `JavaPackage` instance.